calcite-adapter/src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java

```diff
@@ -1124,6 +1119,82 @@ private static String getNullCheckOperator(
+                inputs.add(new Operand(String.valueOf(Integer.MAX_VALUE)));
+            }
+            return new Operand("{$substrCP: [" + Util.commaList(inputs) + "]}");
+        }
+
+        private static Operand getMongoAggregateForConcatOperator(
+                final RexCall call,
+                final List<Operand> strings) {
+            // If using CONCAT function instead of operator, convert any null values to empty string.
+            final List<String> inputs =
+                    strings.stream()
+                            .map(string ->
+                                    call.getOperator() == SqlLibraryOperators.CONCAT_FUNCTION
+                                            ? "{\"$ifNull\": [" + string + ", \"\" ]}"
+                                            : string.toString())
+                            .collect(Collectors.toList());
+            return new Operand("{\"$concat\": [" + Util.commaList(inputs) + "]}");
+        }
+        private static Operand getMongoAggregateForPositionStringOperator(
+                final RexCall call,
+                final List<Operand> strings) {
+            final List<String> args = new ArrayList<>();
+            final StringBuilder operand = new StringBuilder();
+            final StringBuilder finish = new StringBuilder();
+            // Comparison is case-insensitive so convert both strings to same case.
+            // Note also that argument order in $indexOfCP needs to be string, substring, [startIndex].
+            args.add("{\"$toLower\":" + strings.get(1) + "}");
+            args.add("{\"$toLower\":" + strings.get(0) + "}");
+            // Check if either string is null.
+            operand.append("{\"$cond\": [" + RexToMongoTranslator.getNullCheckExpr(strings) + ", ");
+            // Add starting index if any.
+            if (strings.size() == 3) {
+                args.add("{\"$subtract\": [" + strings.get(2) + ", 1]}"); // Convert to 0-based.
+                operand.append("{\"$cond\": [{\"$lte\": [" + strings.get(2) + ", 0]}, 0, "); // Check if 1-based index > 0.
+                finish.append("]}");
+            }
+            // Convert 0-based index to 1-based.
+            operand.append("{\"$add\": [{\"$indexOfCP\": [" + Util.commaList(args) + "]}, 1]}");
+            operand.append(finish);
+            operand.append(", null ]}");
+            // Return 1-based index when string is found.
+            // Returns null if either base string or substring is null.
+            // Returns 0 if substring cannot be found or index (if any) is null or non-positive.
+            return new Operand(operand.toString());
+        }
+
+        private static Operand getMongoAggregateForStringOperator(
+                final RexCall call,
+                final List<Operand> strings) {
+            // Add a null check since these operators do not handle null or missing expressions correctly.
+            return new Operand("{\"$cond\": ["
```

Comment:
Needed to add null check to these operators ($toUpper, $toLower, $strLenCP) since their behaviour was either to return empty string or throw error.